node_modules/col-assets/lib/api.js

```diff
@@ -1711,15 +1711,19 @@ var like = module.exports.like = function(ctx, assetId, like, callback) {
           // Add the like / dislike activity
           var type = like ? 'like' : 'dislike';
 
-          ActivitiesAPI.createActivity(ctx.course, ctx.user, type, asset.id, CollabosphereConstants.ACTIVITY.OBJECT_TYPES.ASSET, null, null, function(err) {
+          ActivitiesAPI.createActivity(ctx.course, ctx.user, type, asset.id, CollabosphereConstants.ACTIVITY.OBJECT_TYPES.ASSET, null, null, function(err, likeActivity) {
             if (err) {
               return callback(err);
             }
+            if (!likeActivity) {
+              log.error({'user': ctx.user, 'assetId': asset.id}, 'Failed to retrieve newly created like/dislike activity');
+              return callback({'code': 500, 'msg': 'Failed to retrieve newly created like/dislike activity'});
+            }
```

Comment:
@johncrossman Here and elsewhere, we're now fetching the `id` property of the newly created activity, which we weren't doing before. So this code path just amounts to a null check so the process doesn't crash if something truly unexpected happens. It is (I hope) very unlikely that it would ever be triggered.